Send logs from subprocess to file instead of pipe.

maniarathi opened the pull request.

parthshahva commented on tests/unit/docker_images/test_validator_harness.py:
```diff
@@ -35,7 +35,13 @@ def setUp(self):
 
+        # A file will be created while the subprocess is running under the name SUBPROCESS_LOG_FILE. This will have
+        # to be cleaned up in tearDown once each test has run.
+        self.log_filename = "SUBPROCESS_LOG_FILE"
```
The log_filename should probably include something unique about the request? Many validators will be running on a single machine

parthshahva commented on upload/docker_images/validator/validator_harness.py:
```diff
@@ -112,20 +112,19 @@ def _run_validator(self):
             'exception': None
         }
         try:
+            _subprocess_filename = "SUBPROCESS_LOG_FILE"
```
The log_filename should probably include something unique about the request? Many validators will be running on a single machine and would stamp on each others log files

maniarathi commented:
Ok, addressed your comments and incremented the version but I think I will need to pair with you next week to figure out how actually publish to docker hub.

sampierson commented:
The original impetus for this ticket was to be able to ssh into instances that appeared to be hung doing validation and see what was going on.
I think saving logs off-box is a separate requirement.  If validation completes, logs are saved in our database.  So this would only cover the case when validation aborted but the harness didn't catch it for some reason.  Not sure how narrow/wide that edge case is.

maniarathi commented:
Woohoo! Fixed the errors and successfully viewed the print statements from the subprocess in file.

maniarathi commented:
Pushed to docker hub.
